pkg/asset/store.go

```diff
@@ -24,6 +32,36 @@ func (s *StoreImpl) Fetch(asset Asset) (*State, error) {
 	return s.fetch(asset, "")
 }
 
+// Load retrieves the state from the state file present in the given directory
+// and returns the assets map
+func (s *StoreImpl) Load(dir string) (map[Asset]*State, error) {
+	path := filepath.Join(dir, stateFileName)
+	assets := make(map[Asset]*State)
+	data, err := ioutil.ReadFile(path)
+	if err != nil {
+		return nil, err
+	}
+	err = yaml.Unmarshal(data, &assets)
+	return assets, err
+}
+
+// Save dumps the entire state map into a file
+func (s *StoreImpl) Save(dir string) error {
+	data, err := yaml.Marshal(&s.assets)
```

Comment:
Let's use JSON instead. Go's YAML implementation has a lot of trouble serializing and then deserializing accurately and YAML is a little too inviting for end users.